docprompt/schema/metadata.py

```diff
@@ -118,23 +131,19 @@ def __len__(self):
     def __getattr__(self, name):
         if self._is_field_typed():
             return super().__getattr__(name)
-        else:
-            return self.extra.get(name)
+
+        if name.startswith("_"):
+            return super().__getattr__(name)
+
+        return self.extra.get(name)
```

Comment:
This should catch a KeyError and reraise an attribute error to reproduce the same functionality as attribute lookup

Reply:
I like that, much cleaner